source/core/GenericFunctions.cpp

```diff
@@ -531,6 +544,15 @@ InstructionCore* EmitGenericUnOpInstruction(ClumpParseState* pInstructionBuilder
 
     InstructionCore* pInstruction = nullptr;
     switch (destType->BitEncoding()) {
+        case kEncoding_Variant:
```

Comment:
ah, good catch. Missed to check if it is "Convert" before emitting "DataToVariant". I think we got lucky because other unary operations throw SA error and we may never hit this except "Convert". Fixing it now to add the check. Thanks.